src/databricks/labs/ucx/framework/utils.py

```diff
@@ -7,25 +7,18 @@
 _allowed_object_chars = set(string.ascii_letters + string.digits + '_')
 
 
-def escape_sql_identifier(path: str, optional: bool | None = True) -> str:
+def escape_sql_identifier(path: str) -> str:
```

Comment:
@larsgeorge-db and @nfx : Do you recall why this was optional in the first place?

In retrospect, I conclude it is confusing to have the optional flag as (new) contributors read/write code like `self._exec(f"TRUNCATE TABLE {escape_sql_identifier(full_name)}")` expecting the SQL identifier will be escaped while it is not. Optionally escaping the sql identifier is more clear outside the function:

```python
if should_escape_sql_identifier:
    full_name = escape_sql_identifier(full_name)
```